TASVideos.WikiEngine/NodeImplementations/Element.cs

```diff
@@ -43,24 +63,38 @@ public Element(int charStart, string tag)
 			throw new InvalidOperationException("Unsupported tag!");
 		}
 
-		CharStart = charStart;
+		_charRange = range;
 		Tag = tag;
+		Attributes = attributes.ToDictionary();
+		Children = children.ToList();
 	}
 
-	public Element(int charStart, string tag, IEnumerable<INode> children)
-		: this(charStart, tag)
-	{
-		Children.AddRange(children);
-	}
+	public Element(
+		int charStart,
+		string tag,
+		IEnumerable<KeyValuePair<string, string>> attributes,
+		IEnumerable<INode> children)
+			: this((charStart, default), tag, attributes, children) { }
 
-	public Element(int charStart, string tag, IEnumerable<KeyValuePair<string, string>> attributes, IEnumerable<INode> children)
-		: this(charStart, tag, children)
-	{
-		foreach (var kvp in attributes)
-		{
-			Attributes.Add(kvp.Key, kvp.Value);
-		}
-	}
+	public Element(
+		StringIndices range,
+		string tag,
+		IEnumerable<KeyValuePair<string, string>> attributes,
+		params INode[] children)
+			: this(range, tag, attributes, children: children.AsEnumerable()) { }
+
+	public Element(
+		int charStart,
+		string tag,
+		IEnumerable<KeyValuePair<string, string>> attributes,
+		params INode[] children)
+			: this((charStart, default), tag, attributes, children: children.AsEnumerable()) { }
+
+	public Element(StringIndices range, string tag)
+		: this(range, tag, [], children: (IEnumerable<INode>)[]) { }
```

Comment:
Does the use of `[]` here automatically pool a single Enumerable.Empty or are we making new objects every time?

Reply:
It just uses [Array.Empty](https://sharplab.io/#v2:D4AQTAjAsAUCAMACEEAsBuWsBiB7XAdACoCmAzgC4AUA2gLoCUmMsAlgHYUkBO7AhgBtkEAGzIwiPLlgBvWIgXIAzMLEhUiUpSoolAHgDSJAJ4A1QQFcSABT6tuelPAA0w+AD53iAGb4G8xTkYRUQAX1hwmCA===).